de-lib/src/main/java/org/iplantc/de/apps/widgets/client/view/editors/AppStepResourcesViewImpl.java

```diff
@@ -0,0 +1,169 @@
+        final long diskSpaceEditorMin = (minDiskSpace != null && minDiskSpace > 0) ? minDiskSpace : ONE_GB;
+        buildResourceSizeLimitList(minDiskSpaceEditor, ONE_GB, diskSpaceEditorMin, deProperties.getToolsMaxDiskLimit());
+
+        final Long memoryLimit = limits.getMemoryLimit();
+        final Long minMemoryLimit = limits.getMinMemoryLimit();
+        final long memoryEditorMin = minMemoryLimit != null ? minMemoryLimit : 2 * ONE_GB;
+        final long memoryEditorMax =
+                (memoryLimit != null && memoryLimit > 0) ? memoryLimit : deProperties.getToolsMaxMemLimit();
+        buildResourceSizeLimitList(memory, 2 * ONE_GB, memoryEditorMin, memoryEditorMax);
+
+        final Double maxCPUCores = limits.getMaxCPUCores();
+        final Double minCPUCores = limits.getMinCPUCores();
+
+        final double cpuEditorMin = (minCPUCores != null && minCPUCores > 1) ? Math.round(minCPUCores) : 1;
+        final double cpuEditorMax =
+                (maxCPUCores != null && maxCPUCores > 0) ? Math.round(maxCPUCores) : deProperties.getToolsMaxCPULimit();
+        buildResourceDoubleLimitList(minCPUCoresEditor, 1, cpuEditorMin, cpuEditorMax);
+
+        setWidget(uiBinder.createAndBindUi(this));
+
+        editorDriver.initialize(this);
+
+        final Splittable json = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(limits));
+        final AppTemplateStepRequirements requirements =
+                AutoBeanCodex.decode(factory, AppTemplateStepRequirements.class, json).as();
+
+        editorDriver.edit(requirements);
+    }
+
+    private void buildResourceSizeLimitList(SimpleComboBox<Long> limitSelectionList,
+                                            long startingLimit,
+                                            long minLimit,
+                                            long maxLimit) {
+        limitSelectionList.add((long)0);
+
+        long resourceLimit = startingLimit;
+        while (resourceLimit <= maxLimit) {
+            if (resourceLimit >= minLimit) {
+                limitSelectionList.add(resourceLimit);
+            }
+
+            resourceLimit *= 2;
+        }
+    }
+
+    private void buildResourceDoubleLimitList(SimpleComboBox<Double> limitSelectionList,
+                                              double startingLimit,
+                                              double minLimit,
+                                              double maxLimit) {
+        limitSelectionList.add((double)0);
+
+        double resourceLimit = startingLimit;
+        while (resourceLimit <= maxLimit) {
+            if (resourceLimit >= minLimit) {
+                limitSelectionList.add(resourceLimit);
+            }
+
+            resourceLimit *= 2;
+        }
+    }
+
+    private SimpleComboBox<Long> createDataSizeComboBox() {
+        final SimpleComboBox<Long> resourceSizeSimpleComboBox =
+                new SimpleComboBox<>(size -> Math.round((float)(size * 10 / ONE_GB)) / 10 + " GB");
```

Comment:
I saw a similar equation like this elsewhere, but I'm not sure we need `* 10` and then also later `/ 10`?

Reply:
This is to ensure the label can display a decimal for partial GB values with `0.1` precision.

For example, if the size were 1,632,087,572 bytes, it should display as "1.5 GB". If we didn't multiply the integer before rounding (which requires dividing after rounding), then `Math.round(1632087572 / ONE_GB) + " GB"` would give us `2 GB`. But `Math.round(1632087572 * 10 / ONE_GB) / 10` gives us the proper `1.5 GB`.

Similarly, if we wanted an extra precision, then we could do `Math.round(1632087572 * 100 / ONE_GB) / 100` to display `1.52 GB`.

Reply:
I see - thanks for the explanation!

Reply:
Heh, I wish `Math.round` just supported rounding to decimals. Maybe it would be good to separate out the scaling factor/precision into a constant? Could then do something like:

```
DISPLAY_PRECISION = 0.1
Math.round((float)(size /DISPLAY_PRECISION)) * DISPLAY_PRECISION + " GB")
```

or call it something like `SCALING_FACTOR` and set it to 10. Not sure which of those ways would be clearer/work better.

(something like `size * (1/DISPLAY_PRECISION)` could also work if dividing by 0.1 is too weird)

Reply:
> Heh, I wish `Math.round` just supported rounding to decimals. Maybe it would be good to separate out the scaling factor/precision into a constant?

I realized that I was casting ints to floats in the wrong places, so this wasn't even working as inteded before 😖
I've since fixed it for this PR: `(float)Math.round((float)(size * 10) / ONE_GB) / 10`

I've seen a few examples of formatting integers this way for javascript code, so I didn't think to clarify what it was doing before, but now I've also added a comment explaining what this conversion is doing.

When this is converted into React, then I think we can use the [numeral](http://numeraljs.com/#format-bytes) package to format these display labels.